test/evp_test.c

```diff
@@ -1811,9 +1811,10 @@ static int encode_test_run(EVP_TEST *t)
     if (expected->encoding == BASE64_CANONICAL_ENCODING) {
         EVP_ENCODE_CTX *encode_ctx;
 
-        if (!TEST_ptr(encode_ctx = EVP_ENCODE_CTX_new())
-                || !TEST_ptr(encode_out =
-                        OPENSSL_malloc(EVP_ENCODE_LENGTH(expected->input_len))))
+        if (!TEST_ptr(encode_out =
+                      OPENSSL_malloc(EVP_ENCODE_LENGTH(expected->input_len))))
+            goto err;
+        if (!TEST_ptr(encode_ctx = EVP_ENCODE_CTX_new()))
             goto err;
```

Comment:
Although swapping the order order of the tests formally solves the leak, it appears to me a little bit over-sophisticated and requires a second thought. I would prefer when `encode_ctx` and `decode_ctx` would be treated analogously, i.e. they should both be declared and NULL-initialized at function scope  (at line 1804)

https://github.com/openssl/openssl/blob/4c66c66c9e2497633102515ce30e217a9f13ed55/test/evp_test.c#L1804

and cleaned up at the `err:` label.

https://github.com/openssl/openssl/blob/4c66c66c9e2497633102515ce30e217a9f13ed55/test/evp_test.c#L1866

Reply:
In particular, this hunk can be reverted if `decode_ctx` is declared at function scope.

Reply:
Maybe it also makes sense to join 4c66c66c9e2 and f27d4e763d8 into a single commit?

Reply:
> 
> Maybe it also makes sense to join [4c66c66](https://github.com/openssl/openssl/commit/4c66c66c9e2497633102515ce30e217a9f13ed55) and [f27d4e7](https://github.com/openssl/openssl/commit/f27d4e763d8baf2fd22d1523dda8321a01440b57) into a single commit?

these commit are not fixing the same type of Coverity diag message, that's why I don't squash them.

Reply:
I squash the 2 "  Fix CID 13xxxx: Resource leak in test/evp_test.c " in one commit, and reorder them a bit .